File: examples/java-spring-boot/src/test/java/dev/httptape/demo/TestApplication.java
package dev.httptape.demo;

import org.springframework.boot.SpringApplication;

/**
 * Dev runner that boots the application with httptape served via Testcontainers.
 *
 * <p>Run this class directly (IntelliJ: right-click &rarr; Run/Debug) or via Maven:
 * <pre>{@code ./mvnw spring-boot:test-run}</pre>
 *
 * <p>The app starts at {@code http://localhost:8080} with a real httptape container
 * serving fixtures in the background. Exit the app to tear down the container.
 */
public class TestApplication {

    public static void main(String[] args) {
        SpringApplication.from(Application::main)
                .with(TestcontainersConfig.class)
                .run(args);
    }
}

File: examples/kotlin-ktor-koog/src/test/kotlin/dev/httptape/demo/TestApplication.kt
package dev.httptape.demo

import io.ktor.server.engine.*
import io.ktor.server.netty.*

/**
 * Dev runner that boots the application with httptape served via Testcontainers.
 *
 * Run this class directly (IntelliJ: right-click -> Run) or via Gradle:
 * ```
 * ./gradlew testRun
 * ```
 *
 * The app starts at `http://localhost:8080` with a real httptape container
 * serving fixtures in the background. Exit the process to tear down the
 * container.
 */
fun main() {
    val baseUrl = HttptapeContainer.baseUrl
    embeddedServer(Netty, port = 8080) {
        configureApp(
            openAiBaseUrl = baseUrl,
            openAiApiKey = "sk-test-key",
            weatherBaseUrl = baseUrl
        )
    }.start(wait = true)
}
